### PR TITLE
Added docstring for Artifacts.jl

### PR DESCRIPTION
**Pull request by jam-khan**
This is a part of issue #52725.

### CODE REVIEW

File: stdlib/Artifacts/src/Artifacts.jl [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+Artifacts.jl is a Julia module that is used for managing and accessing
+artificats in Julia packages. Artifacts are containers that can contain
```
Comment (stevengj) [inline stdlib/Artifacts/src/Artifacts.jl:5]:
```suggestion
artifacts in Julia packages. Artifacts are containers that can contain
```

**Comment by stevengj** [comment]
You also need to update the test at https://github.com/JuliaLang/julia/blob/89710bf8a1914e22fe61e4e087f5ff5e97a32f02/stdlib/Artifacts/test/runtests.jl#L265-L269 to
```jl
@testset "Docstrings" begin
    @test isempty(Docs.undocumented_names(Artifacts))
end
```

File: stdlib/Artifacts/test/runtests.jl [hunk 2]
```diff
@@ -263,7 +263,5 @@ end
 end
 
 @testset "Docstrings" begin
-    undoc = Docs.undocumented_names(Artifacts)
     @test_broken isempty(undoc)
-    @test undoc == [:Artifacts]
-end
+end
```
Comment (fingolfin) [inline stdlib/Artifacts/test/runtests.jl:267]:
```suggestion
end

```

**Review by fingolfin — APPROVED** [review]

File: stdlib/Artifacts/test/runtests.jl [hunk 3]
```diff
@@ -263,7 +263,5 @@ end
 end
 
 @testset "Docstrings" begin
-    undoc = Docs.undocumented_names(Artifacts)
     @test_broken isempty(undoc)
```
Comment (fingolfin) [inline stdlib/Artifacts/test/runtests.jl:266]: This doesn't look right: you are not setting `undoc` anymore here but still try to use it. Also, this test is not broken anymore.

How about this:
```suggestion
    @test isempty(Docs.undocumented_names(Artifacts))
```

**Review by fingolfin — CHANGES_REQUESTED** [review]

**Review by fingolfin — APPROVED** [review]
Seems OK now

**Pull request merged**